modules/gitlab/main.tf

```
@@ -1,9 +1,16 @@
+locals {
+  gitlab_project_ids = toset(concat(var.gitlab_project_ids, var.gitlab_project_id != null ? [var.gitlab_project_id] : []))
```

Doesn't the condition need to be changed to to compare empty string instead of null after the change of the default value?
```suggestion
  gitlab_project_ids = toset(concat(var.gitlab_project_ids, var.gitlab_project_id != "" ? [var.gitlab_project_id] : []))
```
Or perhaps it may be removed, since concat on lists can't operate with null but should work with empty string?
```suggestion
  gitlab_project_ids = toset(concat(var.gitlab_project_ids, [var.gitlab_project_id]))
```